class-08-crud-a-resource/pair-assignment/mikleane-justin08/scripts/article.js

```diff
@@ -0,0 +1,162 @@
+(function(module) {
+  function Article (opts) {
+    // DONE: Convert property assignment to Functional Programming style. Now, ALL properties of `opts` will be
+    // assigned as properies of the newly created article object.
+    Object.keys(opts).forEach(function(e, index, keys) {
+      this[e] = opts[e];
+    },this);
+  }
+
+  Article.all = [];
+
+  Article.prototype.toHtml = function() {
+    var template = Handlebars.compile($('#article-template').text());
+
+    this.daysAgo = parseInt((new Date() - new Date(this.publishedOn))/60/60/24/1000);
+    this.publishStatus = this.publishedOn ? 'published ' + this.daysAgo + ' days ago' : '(draft)';
+    this.body = marked(this.body);
+
+    return template(this);
+  };
+
+  // TODO: Set up a DB table for articles.
+  Article.createTable = function(callback) {
+    webDB.execute('CREATE TABLE IF NOT EXISTS articles(id INTEGER, title TEXT, category TEXT, author TEXT, authorUrl TEXT, publishedOn DATE, body TEXT)',
```

Comment:
I would also suggest putting `PRIMARY KEY` after `id INTEGER` which will designate that column as the unique id for the table. This overrides the default `rowid`, so you can decide what your unique column will be.